### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Unlie9**
pls, for sure i dont wanna to change logic in get_products_with_pagination, because its will be unreal :)

### CODE REVIEW

File: app/parse.py [hunk 1]
```diff
@@ -15,8 +47,133 @@ class Product:
...
+        rating=rating,
+        num_of_reviews=int(product_soup.select_one(
+            ".ratings > p.review-count"
+        ).text.split()[0]),
+    )
+
+
+def get_single_page_products(
+        page_soup: BeautifulSoup, is_paginated: bool = False
+) -> List[Product]:
```
Comment (maxgerman) [inline app/parse.py]: nowadays you could use `list[Product]` without any imports

File: app/parse.py [hunk 2]
```diff
@@ -15,8 +47,133 @@ class Product:
...
+    return all_products
+
+
+def write_products_to_csv(
+        products: List[Product], product_path: str
+) -> None:
+
+    with open(product_path, "w") as file:
+        writer = csv.writer(file)
+        writer.writerow(PRODUCT_FIELDS)
```
Comment (maxgerman) [inline app/parse.py]: the header shouldn't be written to the file multiple times

File: app/parse.py [hunk 3]
```diff
@@ -15,8 +47,133 @@ class Product:
...
 def get_all_products() -> None:
-    pass
+    write_products_to_csv(get_products(HOME_URL), HOME_CSV_PATH)
+    write_products_to_csv(get_products(COMPUTERS_URL), COMPUTERS_CSV_PATH)
+    write_products_to_csv(get_products(PHONES_URL), PHONES_CSV_PATH)
+
+    driver = webdriver.Chrome()
+    try:
+        write_products_to_csv(
+            get_products_with_pagination(driver, LAPTOPS_URL), LAPTOPS_CSV_PATH
```
Comment (maxgerman) [inline app/parse.py]: the pagination detection should be automatic

**Review by maxgerman — CHANGES_REQUESTED** [review]
Good job!
See some comments.

**Comment by maxgerman** [comment]
should be automated, adapt the function that gets the products form the page (have "more" button -> click in a loop)

**Review by maxgerman — CHANGES_REQUESTED** [review]
no rest for the wicked :)

**Comment by Unlie9** [comment]
> no rest for the wicked :)

😌

**Review by DevSpaciX — APPROVED** [review]

**Pull request closed without merging**